lib/src/dartdoc.dart

```diff
@@ -60,14 +88,17 @@ class DartdocFileWriter implements FileWriter {
 
   @override
   void write(String filePath, String content, {Warnable? element}) {
+    final bytes = utf8.encode(content);
```

Comment:
I imagine this might increase memory pressure, even though it's local... we can verify whether it is or not with DevTools.

Reply:
`File.writeAsStringSync` will also call `writeAsBytesSync(encoding.encode(contents), ...` internally, so it wouldn't really change much, the encoding happens a few instructions earlier, and maybe the GC of this object may trigger slightly later, but otherwise I don't think it will affect much.

Reply:
It would be interesting to see if writing `utf8` as a stream is faster, but that would probably require a lot of changes to the code -- if we don't write files concurrently it might not matter.

Reply:
Ah I see. This is good then.

> It would be interesting to see if writing `utf8` as a stream is faster

yeah I think for large packages we do spend a lot of time just pouring bytes onto disk, and I'd be game for any way to speed that up :D

Reply:
IIRC I had experimented with tuning the writes at one point, making them happening async and also caching them and do writes after all the work was done in dartdoc. Neither proved significant change (but it revealed a costly operation in the fs abstraction which got fixed :). Unless we have a strong suspect where we have a bottleneck here, I wouldn't spend much more time on it.